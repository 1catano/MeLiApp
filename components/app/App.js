import React from "react";
import PropTypes from "prop-types";
import { useRouter } from "next/router";
import { I18nextProvider } from "react-i18next";
import i18n from "../../translations/i18n";
import SearchBox from "../search-box/SearchBox";

import sty from "./app.module.scss";

const AppComponent = (props) => {
    const cssClass = `${sty["ml-app__main"]} ${props.fullHeight ? sty["ml-app__main--full-height"] : ""}`;
    const router = useRouter();
    const handleQuery = (event) => {
        if (event.keyCode == 13) {
            const query = event.target.value;
            router.push(`/items/?q=${query}`, undefined, { shallow: false });
        }
    };
    return (
        <I18nextProvider i18n={i18n}>
            <div className={cssClass}>
                <SearchBox handleQuery={handleQuery} />
                <div className="container">{props.children}</div>
            </div>
        </I18nextProvider>
    );
};

AppComponent.AppComponent = {
    children: PropTypes.node,
    fullHeight: PropTypes.bool,
};

AppComponent.defaultProps = {
    children: null,
    fullHeight: true,
};

export default AppComponent;
