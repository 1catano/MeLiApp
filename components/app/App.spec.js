import React from "react";
import { shallow } from "enzyme";
import App from "./App";
describe("App snapshot", () => {
    it("take snapshot", () => {
        const component = shallow(<App />);
        expect(component).toMatchSnapshot();
    });
});
