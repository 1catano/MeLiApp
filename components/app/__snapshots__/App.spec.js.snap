// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot render amount and divider 1`] = `ShallowWrapper {}`;

exports[`App snapshot take snapshot 1`] = `ShallowWrapper {}`;
